pkg/plugin/system/system.go

```diff
@@ -174,24 +178,25 @@ func (ps *pluginSystem) commandHooksInterceptor(methodName string, streams iouti
 			return fmt.Errorf("failed to run 'aspect %s' command: %w", cmd.Use, err)
 		}
 
-		// TODO(f0rmiga): test this hook.
 		defer func() {
-			hasErrors := false
 			for node := ps.plugins.head; node != nil; node = node.next {
 				params := []reflect.Value{
 					reflect.ValueOf(isInteractiveMode),
 					reflect.ValueOf(ps.promptRunner),
 				}
 				if err := reflect.ValueOf(node.plugin).MethodByName(methodName).Call(params)[0].Interface(); err != nil {
-					fmt.Fprintf(streams.Stderr, "Error: failed to run 'aspect %s' command: %v\n", cmd.Use, err)
-					hasErrors = true
+					exitErr = &aspecterrors.ExitError{
```

Comment:
So this now propagates plugin errors to cause the interceptor to return an error. Is that correct or should it only print to `Stderror` and do nothing else?

Reply:
Previously this error was only logged, `exitErr` was never assigned to and only came from `next()`. Assuming I'm understanding it correctly...

Previously:
* plugin errors are printed
* plugin errors set `hasErrors = true`
* interceptor errors were assigned `ExitCode = 1` **iff** `hasErrors` was true (<--- that **iff** is a bug imo)

Now:
1. plugin errors are printed
2. plugin errors are recorded and override any errors from `next()` (and override previous plugin errors)
3. all errors are assigned `ExitCode = 1`

I'm not sure which errors should have priority in 2) there though... should errors from `next()` override the plugin errors? (currently this PR does the opposite)